src/main/java/com/vmware/vipclient/i18n/base/cache/MessageCache.java

```diff
@@ -4,18 +4,27 @@
  */
 package com.vmware.vipclient.i18n.base.cache;
 
+import java.util.HashMap;
 import java.util.LinkedHashMap;
 import java.util.Map;
 import java.util.Set;
 
+import com.vmware.vipclient.i18n.VIPCfg;
+import com.vmware.vipclient.i18n.messages.api.url.URLUtils;
+
 public class MessageCache implements Cache {
     private String                           id                  = "cache-default";
 
     private long                             expiredTime         = 864000000;                                       // 240hr
     private long                             lastClean           = System.currentTimeMillis();
 
-    private Map<String, Map<String, String>> cachedComponentsMap = new LinkedHashMap<String, Map<String, String>>();
-
+    private final Map<String, Map<String, String>> cachedComponentsMap = new LinkedHashMap<String, Map<String, String>>();
+    private final Map<String, Map<String, Object>> cacheProperties = new LinkedHashMap<String, Map<String, Object>>();
```

Comment:
If you don't need frequent delete or add data, I am not recommended use LinkedHashMap